.github/workflows/ats.publish.yml

```
@@ -0,0 +1,160 @@
+name: ATS Publish
+
+on:
+  # Manual trigger with dry-run option
+  workflow_dispatch:
+    inputs:
+      dry-run-enabled:
+        description: 'Run npm publish with dry-run flag'
+        required: false
+        type: boolean
+        default: false
+
+  # Release published trigger (GitHub release creation) - only for ATS releases (tags ending with -ats)
+  release:
+    types:
+      - published
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  contracts:
+    name: Publish ATS Contracts
+    runs-on: token-studio-linux-large
+    # Only run if manual trigger OR release tag contains 'ats'/'ATS'
+    if: ${{ github.event_name == 'workflow_dispatch' || contains(github.ref_name, 'ats') || contains(github.ref_name, 'ATS') }}
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@0634a2670c59f64b4a01f0f96f84700a4088b9f0 # v2.12.0
+        with:
+          egress-policy: audit
+
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+
+      - name: Setup NodeJS Environment
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
+        with:
+          node-version: 22.x
+          registry-url: https://registry.npmjs.org
+
+      - name: Create .npmrc file
+        working-directory: packages/ats/contracts
+        run: |
+          cat << 'EOF' > .npmrc
+          //registry.npmjs.org/:_authToken=${NODE_AUTH_TOKEN}
+          EOF
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build ATS Contracts
+        run: npm run ats:contracts:build
+
+      - name: Publish ATS Contracts
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          DRY_RUN: ${{ inputs.dry-run-enabled }}
+        working-directory: packages/ats/contracts
+        run: |
+          PUBLISH_ARGS=("--access=restricted")
+          if [[ "${DRY_RUN}" == "true" ]]; then
+            PUBLISH_ARGS+=("--dry-run")
+            echo "🔍 DRY RUN MODE: Would publish @hashgraph/asset-tokenization-contracts"
+          fi
+
+          if ! npm publish "${PUBLISH_ARGS[@]}"; then
+            echo "❌ Failed to publish package: ${package_name}"
+            echo "📋 Package info:" && cat package.json | jq '.name, .version'
+            exit 1
+          fi
+
+  sdk:
+    name: Publish ATS SDK
+    runs-on: token-studio-linux-large
+    # Only run if manual trigger OR release tag contains 'ats'/'ATS'
+    if: ${{ github.event_name == 'workflow_dispatch' || contains(github.ref_name, 'ats') || contains(github.ref_name, 'ATS') }}
```

This would match any branch or tag including the words:
- stats
- formats
- chats
- nats
- beats
- strats
- statsd
- automats

Assuming the branch/tag format is `<issue_no>-(ats|ATS)-<descriptive-branch-name>` then we should be doing something like this:

```suggestion
    if: ${{ github.event_name == 'workflow_dispatch' || contains(github.ref_name, '-ats-') }}
```

[Github also ignores case when doing comparisons](https://docs.github.com/en/actions/reference/workflows-and-actions/expressions#functions) so the second `contains` check is redundant and should be removed.